tools/clang/lib/SPIRV/LowerTypeVisitor.cpp

```diff
@@ -60,11 +60,19 @@ bool LowerTypeVisitor::visitInstruction(SpirvInstruction *instr) {
   const QualType astType = instr->getAstResultType();
   const SpirvType *hybridType = instr->getResultType();
 
+  spv::ImageFormat format = instr->getImageFormat();
+  // Propagate image format to the result type of load
+  if (instr->getopcode() == spv::Op::OpLoad) {
```

Comment:
this code makes me a bit nervous. Are there other special cases to consider? Either way the propagation of this information should be done inside a separate visitor pass, not inside LowerTypeVisitor.cpp

Reply:
I looked into this a bit more and realized that this Propagation to OpLoad is unnecessary and works fine without it. So basically this special case can and should be removed.

The propagation of the defined type is not a special case, but more like a requirement for all stores and loads with this change (requiring unlimited number of propagations because the variable can be stored multiple times). But legalization in SPIRV-Tools fixes all of those cases, and even this single propagation step does not need to be here in the codegen.